src/core/reactor.cc

```diff
@@ -1775,21 +1775,35 @@ void reactor_backend_epoll::forget(pollable_fd_state& fd) {
 
 pollable_fd
 reactor::posix_listen(socket_address sa, listen_options opts) {
-    file_desc fd = file_desc::socket(sa.u.sa.sa_family, SOCK_STREAM | SOCK_NONBLOCK | SOCK_CLOEXEC, int(opts.proto));
-    if (opts.reuse_address) {
-        fd.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1);
-    }
-    if (_reuseport)
-        fd.setsockopt(SOL_SOCKET, SO_REUSEPORT, 1);
+    if (opts.proto == transport::UNIX) {
```

Comment:
```suggestion
    if (opts.proto.sa_family == AF_UNIX) {
```
how about checking `sa.as_posix_sockaddr.sa_family` ?

Reply:
Existing seastar code usually checks the transport. And the code looks better (to my taste...)

Reply:
again, see https://github.com/ceph/seastar/pull/7#discussion_r312134930